packages/@adobe/spectrum-css-temp/components/calendar/index.css

```diff
@@ -205,64 +216,61 @@ governing permissions and limitations under the License.
     }
   }
 
+  /* hover and selected states */
+  &:after {
+    position: absolute;
+    top: calc(50% - calc(var(--spectrum-calendar-day-width) / 2));
+    left: calc(50% - calc(var(--spectrum-calendar-day-width) / 2));
+
+    display: block;
+
+    height: var(--spectrum-calendar-day-height);
+    width: var(--spectrum-calendar-day-width);
+
+    border-radius: var(--spectrum-calendar-day-width);
+
+    content: '';
+    z-index: -1;
+  }
+
   &.is-range-selection {
     /* overrides to make the cells appear connected */
     margin: var(--spectrum-calendar-day-padding) 0;
     border-width: var(--spectrum-calendar-border-width-reset);
     line-height: var(--spectrum-calendar-day-height);
     border-radius: var(--spectrum-calendar-border-radius-reset);
-
-    /* Todo: Calculate this */
-    width: calc(var(--spectrum-calendar-day-width) + calc(var(--spectrum-calendar-day-padding) * 2));
-
-    &.is-range-start,
-    &.is-range-end,
-    &.is-selection-start,
-    &.is-selection-end {
-      width: calc(var(--spectrum-calendar-day-width) + var(--spectrum-calendar-day-padding));
-    }
-
-    /* hover and selected states */
-    &:after {
-      position: absolute;
-      top: calc(50% - calc(var(--spectrum-calendar-day-width) / 2));
-      left: calc(50% - calc(var(--spectrum-calendar-day-width) / 2));
-
-      display: block;
-
-      height: var(--spectrum-calendar-day-height);
-      width: var(--spectrum-calendar-day-width);
-
-      border-radius: var(--spectrum-calendar-day-width);
-
-      content: '';
-      z-index: -1;
-    }
+    /* this is the computed value of the "auto" margin that the non-selected cells have. */
+    --margin-start: calc((100% - var(--spectrum-calendar-day-width)) / 2);
 
     &.is-range-start,
     &.is-selection-start {
-      padding-inline-end: var(--spectrum-calendar-day-padding);
-      margin-inline-start: var(--spectrum-calendar-day-padding);
       border-start-start-radius: var(--spectrum-calendar-day-width);
       border-end-start-radius: var(--spectrum-calendar-day-width);
+      /* push over by the "auto" margin so the background doesn't extend to the left */
+      margin-inline-start: var(--margin-start);
+      width: calc(100% - var(--margin-start));
 
-      &:before,
-      &:after {
+      &:after,
+      .spectrum-Calendar-dateText {
         inset-inline-start: 0;
+        margin-inline-start: 0;
```

Comment:
I assume the ring isn't supposed to be off center right?
![image](https://user-images.githubusercontent.com/9661127/139754608-9bc7f2a3-7bdb-4122-98c6-c31ef9665e15.png)

Reply:
Fixed